### PR TITLE
fix(policies): avoid torch.compile graph breaks from loss.item()

### PR DESCRIPTION
**Pull request by Daankrol**
## Summary

All native policy models (`pi0`, `pi05`, `act`, `groot`, `smolvla`) call `.item()` on the loss tensor inside `compute_loss()`, which is part of the call chain of the `torch.compile`-wrapped `forward`. `Tensor.item()` forces a host sync that Dynamo cannot trace, so every compiled training step hits a graph break:

```
Graph break from `Tensor.item()`, consider setting:
    torch._dynamo.config.capture_scalar_outputs = True
```

reported originally for `pi05`, but the same pattern exists in all Studio-native policies.

## Fix

- Return the (detached) loss tensor(s) instead of Python floats from `compute_loss()` in each policy's model. `self.log(...)` (Lightning) and the training callbacks (`_extract_loss` in `physicalai.train.callbacks`) already accept tensors directly, so no behavior change for callers — the scalar host sync now happens lazily outside the compiled graph.
- `compute_val_loss()` is **not** part of the compiled `forward` chain (validation bypasses `forward` and calls `compute_val_loss` directly), so its `.item()` calls are left unchanged.
- Updated the `Model.compute_loss` / `compute_val_loss` base signatures (and all overrides) from `dict[str, float]` to `dict[str, Tensor | float]` to reflect that either representation may now be returned.
- pi0's `_forward_loss` also dropped a `.tolist()` call on `loss_per_dim` (same root cause) in favor of a detached tensor.

## Verification

- Reproduced the graph break with `TORCH_LOGS=graph_breaks` on `ACT` with `compile_model=True` before the fix (confirmed graph breaks at each `.item()` call site), and confirmed they are gone after the fix, with no other graph break in the loss path.
- `uv run pytest tests/unit/policies/{test_pi05,test_pi0,test_act,test_groot,test_smolvla,test_eval_loss,test_lerobot}.py` — 270 passed. The 9 failures present are pre-existing on `main` (missing `diffusers` extra for groot DiT tests; unrelated lerobot diffusion-policy failures) and unrelated to this change.
- `prek run` (ruff, ruff-format, mypy, pyrefly, bandit) passes on all touched files.

## Note

Signed commits were requested by repo guidelines, but no GPG/SSH signing key is configured in this environment, so this commit is unsigned. Please let me know if you'd like me to re-sign it before merging.

### CODE REVIEW

File: library/src/physicalai/policies/base/model.py [hunk 1]
```diff
@@ -32,18 +32,23 @@ class Model(nn.Module, ABC):
...
         Args:
             batch: Preprocessed batch dict.
 
         Returns:
             Tuple of (loss tensor with grad, dict with at least a ``"loss"`` key).
+            Values in the dict should generally be left as (detached) tensors
+            rather than converted to Python scalars via ``.item()``: when the
+            model's ``forward`` is wrapped with ``torch.compile``, calling
+            ``.item()`` inside this method forces a host sync and breaks the
+            compiled graph. ``self.log`` accepts tensors directly.
```
[the remark this answers is not in the document]
Comment (Daankrol) [reply library/src/physicalai/policies/base/model.py]: Fixed — reworded to reference the enclosing `Policy` (`LightningModule`)'s `self.log(...)` instead of implying `Model` itself has `.log`. See ce366f27.

File: library/src/physicalai/policies/base/policy.py [hunk 2]
```diff
@@ -244,7 +244,7 @@ def validation_step(self, batch: Gym | Observation, batch_idx: int) -> dict[str,
             return loss
         return self.evaluate_gym(batch, batch_idx, stage="val")
 
-    def compute_val_loss(self, batch: Observation) -> tuple[torch.Tensor, dict[str, float]]:
+    def compute_val_loss(self, batch: Observation) -> tuple[torch.Tensor, dict[str, torch.Tensor | float]]:
```
Comment (sovrasov) [inline library/src/physicalai/policies/base/policy.py:247]: Looks like we would never have floats here, it can be removed to reduce annotation length

**Comment by Daankrol** [comment]
@copilot resolve the merge conflicts in this pull request

**Review by sovrasov — APPROVED** [review]
